ui/app/AppLayouts/Chat/stores/PermissionsStore.qml

```diff
@@ -1,14 +1,65 @@
 import QtQml 2.15
 
+import SortFilterProxyModel 0.2
+import utils 1.0
+
 QtObject {
     id: root
 
     required property string activeSectionId
+    required property string activeChannelId
     required property var chatCommunitySectionModuleInst
 
+    // all permissions model
     readonly property var permissionsModel:
         chatCommunitySectionModuleInst.permissionsModel
 
+    readonly property var becomeMemberPermissionsModel: SortFilterProxyModel {
+        id: becomeMemberPermissionsModel
+        sourceModel: root.permissionsModel
+        function filterPredicate(modelData) {
+            return modelData.permissionType == Constants.permissionType.member
+        }
+        filters: [
+            ExpressionFilter {
+                expression: becomeMemberPermissionsModel.filterPredicate(model)
+            }
+        ]
+    }
+
+    readonly property var viewOnlyPermissionsModel: SortFilterProxyModel {
+        id: viewOnlyPermissionsModel
+        sourceModel: root.permissionsModel
+
+        function filterPredicate(modelData) {
+            return modelData.permissionType == Constants.permissionType.read && root.permissionsModel.belongsToChat(modelData.id, root.activeChannelId)
+        }
+        filters: [
+            ExpressionFilter {
+                expression: {
+                    root.activeChannelId // ensure predicate is re-triggered when activeChannelId changes
+                    viewOnlyPermissionsModel.filterPredicate(model)
+                }
+            }
+        ]
+    }
+
+    readonly property var viewAndPostPermissionsModel: SortFilterProxyModel {
+        id: viewAndPostPermissionsModel
+        sourceModel: root.permissionsModel
+        function filterPredicate(modelData) {
+            return modelData.permissionType == Constants.permissionType.viewAndPost && root.permissionsModel.belongsToChat(modelData.id, root.activeChannelId)
+        }
+        filters: [
+            ExpressionFilter {
+                expression: {
+                    root.activeChannelId // ensure predicate is re-triggered when activeChannelId changes
```

Comment:
do not understand why do we need this. filterPredicate inside contains root.activeChannelId call

Reply:
This is a hack to re-trigger the expression filter when `channelID` changes. Otherwise it doesn't which results in the view model not updating correctly.

/cc @osmaczko

Reply:
Otherwise the `SortFilterProxyModel` will not be refreshed when `activeChannelId` is changed.